grpcbp/client_middlewares_test.go

```diff
@@ -0,0 +1,174 @@
+package grpcbp
+
+import (
+	"context"
+	"encoding/json"
+	"net"
+	"testing"
+	"time"
+
+	pb "github.com/grpc-ecosystem/go-grpc-middleware/testing/testproto"
+	"github.com/opentracing/opentracing-go"
+	"github.com/reddit/baseplate.go/mqsend"
+	"github.com/reddit/baseplate.go/tracing"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/test/bufconn"
+)
+
+func TestMonitorInterceptor(t *testing.T) {
+	// create test server
+	l, _ := initServer(t)
+
+	// create test client with monitoring interceptor
+	conn := initClient(t, l, grpc.WithUnaryInterceptor(
+		MonitorInterceptor(
+			MonitorClientArgs{
+				ServiceSlug: "test",
+			},
+		),
+	))
+
+	// instantiate gRPC client
+	ctx, mmq := initServerSpan(t)
+	client := pb.NewTestServiceClient(conn)
+
+	t.Run("span-success", func(t *testing.T) {
+		_, err := client.Ping(ctx, &pb.PingRequest{})
+		if err != nil {
+			t.Fatal(err)
```

Comment:
at least give it a tiny bit of context, e.g. `Ping: %s`

(same below)

Reply:
Does the context help though if the user needs to go and look at the test at this line anyway? I remember we had a similar discussion about this. I almost feel the test code is easier to read without all the string interpolations to give additional context.

Other than that I am happy to standardize is going forward. How would we do it for instance with:
```go
if err = json.Unmarshal(msg, &trace); err != nil {
			t.Fatal(err)
		}
```
Should it be `Unmarshal: %s` or `json.Unmarshal: %s`? :thinking:

Reply:
I think when it's in a function without `t.Helper()` the line number alone should be enough. But we do need context if it's called inside a function that called `t.Helper()` (as that will remove the line numbers from within that function).

But I won't mind if we decided to force always adding the context to `t.Fatal`.